### PR TITLE
ADAPT-2695 Create Poster component

### PR DESCRIPTION
**Pull request by rebeccahongsf**
# READY FOR REVIEW

# Summary
- Create Poster component

# Needed By (Date)
- tbd

# Urgency
- low

# Steps to Test

1. Pull this PR
2. Run `npm run dev` or view [component on netlify here](https://deploy-preview-50--decanter-react.netlify.app/?path=/story/composite-poster--default)
3. Review code and component display 

# Associated Issues and/or People
- [ADAPT-2695](https://stanfordits.atlassian.net/browse/ADAPT-2695)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: src/Poster/Poster.js
Comment (yvonnetangsu) [inline src/Poster/Poster.js:35]: I think you're totally on the right track. If you worry about spacing above/below the different parts, it's ok to put those in the subcomponents 😄  (<- not sure if that's what you were concerned about). 

FYI (and you probably already know this), the first/last prefixes and the TW classes with [] are not available for this repo sadly 😢 

Let's check with everyone if we want to include background image in this component for now - don't want to scope creep on ya 😂  Please feel free to post to the adapt channel - let's have a discussion on what background color/text colors to include so you don't end up doing more work than needed.

File: src/Poster/Poster.stories.js
Comment (yvonnetangsu) [inline src/Poster/Poster.stories.js:37]: children prefix also not available 😢

File: src/common/util/addBgImage.js
Comment (yvonnetangsu) [inline src/common/util/addBgImage.js:21]: For this repo, totally ok to take out image processing (resize, change jpg quality etc) since these were added to use with Storyblok's image service. For this repo, I think we can leave it up to the user to pass in the image processed on their end.

**Comment by yvonnetangsu** [comment]
@rebeccahongsf Looking better and better. A few ideas - please feel free to push back:

- Let's use the foggy light poster as the first example story, since the white poster blends in with the page background. Also, I suspect a lot of websites have white backgrounds, so having a light grey poster would make it actually look like a poster instead of just a WYSIWYG among the rest of the page content 😄 
- Perhaps some examples with the various CTA buttons in the poster body? (white poster with solid red, black with the ghost button sound pretty good 👻 )
- A couple headline and body copy text size options like we have in SAA site would be nice, in examples totally fine.
- This is from Kerri from previous SAA communications, for `su-card-paragraph` sized body copy, she'd like the line height to be 1.3 (su-leading-snug).
- Let's kill the bottom margin on the last element in the poster.

Thanks in advance!

File: src/Poster/Poster.js
Comment (rebeccahongsf) [reply src/Poster/Poster.js:38]: I had set a default `linear-gradient` for the `gradientOverlay` variable in the event that the developer forgets to pass in a gradient.

Does it make sense to have a gradient option (lever)? 
Alternatively, we could remove the default gradient and have it as an empty string instead. That way it would be completely in the developer's hands to decide whether they want an overlay on their `bgImage`. Thoughts? 😄

File: src/Poster/Poster.js
Comment (rebeccahongsf) [reply src/Poster/Poster.js]: Realized that `first:` and `last:` prefixes do not work within out decanter-react repo for flex styles (margin only but even then it's a bit finicky). 😢  I'll rewrite to include the alignments within the stories and remove the lever. Perhaps we can revisit adding the alignment lever if/when we expand the coverage of `first:` and `last:` prefixes?

Comment (rebeccahongsf) [reply src/Poster/Poster.js]: Or we can keep the layout lever since it's could be seen as image left or center! The body content will be left align unless additional classes are passed to the Content subcomponent.

File: src/Poster/Poster.js
Comment (sherakama) [reply src/Poster/Poster.js]: I was more thinking of an on/off lever for bg gradients vs options. What if the developer wants no gradient? Can you make it easy to remove?

**Review by sherakama — APPROVED** [review]
This is great. Thanks for the work. I refactored the Poster. Image component a bit as the image.src object requirement was too restrictive for using things like Gatsby's image optimization components. I left the background image alone as that is less likely to need a component to optimize the image.

Screen Shot 2021-06-09 at 11 08 28 PM


Screen Shot 2021-06-09 at 11 08 50 PM

File: src/Poster/Poster.js
Comment (sherakama) [inline src/Poster/Poster.js:30]:
```suggestion
      "su-rs-mb-2 su-mx-auto md:su-rs-mr-4 md:su-mb-0 md:su-ml-0";
```

Sorry, no JIT compiler here.

File: src/Poster/Poster.js
Comment (sherakama) [inline src/Poster/Poster.js:109]: Missing props for bgGradient and bgColor

File: src/Poster/Poster.stories.js
Comment (yvonnetangsu) [inline src/Poster/Poster.stories.js:101]: If you could please add a `su-mb-0` to last paragraphs in all the examples, that'd be great.

File: src/Poster/Poster.stories.js
Comment (yvonnetangsu) [inline src/Poster/Poster.stories.js:144]: Looks like Figma image border is a 7px one, but Decanter v7 doesn't have a 7px width currently. I know @sherakama added some recently, so let's go with either 6px or 8px 😄  Although... @sherakama , is there a release that got punched with your border width PR?
https://github.com/SU-SWS/decanter/pull/834/files

**Review by yvonnetangsu — CHANGES_REQUESTED** [review]
Just those few final minor requests, and one more here:

For the "default" (the first example), let's add some classes so the content is center-aligned (it's a bit more commonly used), and let's add a solid red CTA button with it 😄 

After that, then your work is done - great job 👏🏼  And thank you for your help in refactoring too, @sherakama !

**Comment by yvonnetangsu** [comment]
@rebeccahongsf One minor note - if you could use the "large" (or was it "big") CTA button `size` prop for the poster examples, that'd be awesome 🙏🏼

File: src/Poster/Poster.stories.js
Comment (rebeccahongsf) [reply src/Poster/Poster.stories.js:144]: The latest release does not seem to include the border-width 😢  Currently the available options are
```
.su-border-0{
  border-width: 0
}

.su-border-2{
  border-width: 2px
}

.su-border-3{
  border-width: 3px
}

.su-border{
  border-width: 1px
}
```

Comment (sherakama) [reply src/Poster/Poster.stories.js:144]: Send in a PR to add border-7 and add the `alpha` label to it. That should get you what you need.

**Comment by sherakama** [comment]
@yvonnetangsu I think this is GTG now too. Anything left?

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu I think this is GTG now too. Anything left?

Very very close - if we could center align the "default" example and add a CTA button, and use the big CTA button for the last teal poster example then we're GTG 😄

**Comment by rebeccahongsf** [comment]
@yvonnetangsu examples have been updated! 😄  Let me know if any other adjustments or examples are needed

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu examples have been updated! 😄 Let me know if any other adjustments or examples are needed

Thank you~ If you could please use the large/big button size for the black and lagunita posters, then we're GTG. We can deal with the border width on the image later - I think Shea suggested open a PR in Decanter v7 to add border width 7px, but that can totally wait 😸

File: src/Poster/Poster.stories.js
Comment (yvonnetangsu) [inline src/Poster/Poster.stories.js:153]:
```suggestion
        size="big"
```

**Review by yvonnetangsu — APPROVED** [review]
GTG - thank you for the changes! Nice work 🎉 😸  We can add that image border width change later after the Decanter PR get merged. Thank you!

**Comment by yvonnetangsu** [comment]
@sherakama Got this email - "Auto Release: All jobs have failed", but it says all 11 checks passed. Any ideas? 🤔

**Comment by sherakama** [comment]
Looks like the release failed to build https://github.com/SU-SWS/decanter-react/runs/2824551382?check_suite_focus=true 

Could be missing export or missing `npm run package`

**Pull request merged**
